### ISSUE TITLE
Content Assist doesn't reliably work in generator functions

### ISSUE THREAD

**Issue by szarnekow** [eclipse/n4js#39]
The following code doesn't content assist the members of req:

```javascript
class A {}
export public function* foo(req: A) { // removing export public fixes the problem 
    yield 5; // removing 5 fixes the problem
    req. // <-- content assist here ctrl+space
}
```

Probably a problem with our parameterized rules. In some cases, a wrong method name is generated. This rule name is created with information from the generated (Content Assist) parser, cf. https://github.com/eclipse/n4js/blob/master/plugins/org.eclipse.n4js.ui/src-gen/org/eclipse/n4js/ui/contentassist/antlr/N4JSParser.java#428
This leads to a non-existing method name: java.lang.NoSuchMethodException: eu.numberfour.n4js.ui.contentassist.CustomN4JSParser$CustomInternalN4JSParser.norm3_AssignmentExpression__Group_3__0()

Potential solutions:

1)    fix generated code (Xtext bug)
2)    avoid usage of parameterized rules by moving the yield/non-yield check to the validator (ASTStructureValidator)

However this may only be a consequential or different error, accidentally revealed by this problem.

(initial reporter: @mmews-n4 , comments: @jpilgrim

**Comment by szarnekow** [comment]
I prefer solution (1)

**Comment by szarnekow** [comment]
Turns out that the NoSuchMethodExceptions are unfortunate, but did not have any impact on this bug.

**Comment by arrowplum** [comment]
Was this fixed? Is it in the n4js build? I am confused at the status here.

**Comment by szarnekow** [comment]
Not merged yet. Someone with write-access needs to press the button.

**Comment by yoosiba** [comment]
merged with 89bcf0b114041d0796fd4fc3738620c2c6c33623
